python/setup.py

```diff
@@ -80,41 +80,55 @@ def run(self):
         # We also need to install pyarrow along with Ray, so make sure that the
         # relevant non-Python pyarrow files get copied.
         pyarrow_files = []
-        for (root, dirs, filenames) in os.walk("./ray/pyarrow_files/pyarrow"):
+        for (root, dirs,
+             filenames) in os.walk("python/ray/pyarrow_files/pyarrow"):
             for name in filenames:
-                pyarrow_files.append(os.path.join(root, name))
+                pyarrow_files.append(name)
+
+        for filename in pyarrow_files:
+            self.move_file(filename, filename)
 
-        files_to_include = ray_files + pyarrow_files
+        files_to_include = ray_files
 
         # Copy over the autogenerated flatbuffer Python bindings.
         for directory in generated_python_directories:
-            for filename in os.listdir(directory):
+            for filename in os.listdir(os.path.join(SOURCE_PREFIX, directory)):
                 if filename[-3:] == ".py":
                     files_to_include.append(os.path.join(directory, filename))
 
         for filename in files_to_include:
-            self.move_file(filename)
+            source = os.path.join(SOURCE_PREFIX, filename)
+            self.move_file(source, filename)
+
+        # The UI files are mandatory if the INCLUDE_UI environment variable
+        # equals 1. Otherwise, they are optional.
+        for filename in ray_ui_files:
+            if "INCLUDE_UI" in os.environ and os.environ["INCLUDE_UI"] == "1":
+                self.move_file(filename, filename)
+            else:
+                self.move_file(filename, filename, optional=True)
 
         # Try to copy over the optional files.
         for filename in optional_ray_files:
-            try:
-                self.move_file(filename)
-            except Exception:
-                print("Failed to copy optional file {}. This is ok."
-                      .format(filename))
+            source = os.path.join(SOURCE_PREFIX, filename)
+            self.move_file(source, filename, optional=True)
 
-    def move_file(self, filename):
+    def move_file(self, source, filename, optional=False):
```

Comment:
The `optional` argument does not appear to be used.

Reply:
It's used while copying the webui files.

Reply:
I understand now, you meant the method is not using it. I changed it now.